### ISSUE TITLE
Burn: Prerequisite bootstrapper application with /norestart option gets stuck in an infinite installation loop

### ISSUE THREAD

**Issue by dsed** [wixtoolset/issues#5286]
Please provide answers to the following questions to help us narrow down, reproduce, and fix the problem. Fill out one section and delete the others.
## Bugs

If this issue is a bug:
- Which version of WiX are you building with?

> 3.9.1208.0
> 3.10.2.2516
- Which version of Visual Studio are you building with (if any)?

> Visual Studio 2013 Update 5
- Which version of .NET are you building with?

> .NET 4.5.2
- If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> Windows Server 2012 Standard
- Describe the problem and the steps to reproduce it.

> The prerequisite bootstrapper application gets stuck in an infinite installation loop when /norestart is specified on the command line and the .Net prerequisite package requests a machine restart after upgrading .Net 4.5.1 to .Net 4.5.2.

After the initial .Net update burn reports

```
[0990:0268][2016-05-03T13:46:56]i399: Apply complete, result: 0x0, restart: Initiated, ba requested restart:  No
[0990:0268][2016-05-03T13:46:57]i500: Shutting down, exit code: 0x669
[0990:0268][2016-05-03T13:46:57]i000: The prerequisites were successfully installed. The bootstrapper application will be reloaded.
[0990:0268][2016-05-03T13:46:57]i006: Bootstrapper application requested to be reloaded.
[0990:0268][2016-05-03T13:46:58]i000: Loading prerequisite bootstrapper application because managed host could not be loaded, error: 0x80131700.
```

Burn starts all over and over again always resulting in

```
[0990:0268][2016-05-03T13:53:26]i399: Apply complete, result: 0x0, restart: None, ba requested restart:  No
[0990:0268][2016-05-03T13:53:26]i500: Shutting down, exit code: 0x0
[0990:0268][2016-05-03T13:53:26]i000: The prerequisites were successfully installed. The bootstrapper application will be reloaded.
[0990:0268][2016-05-03T13:53:26]i006: Bootstrapper application requested to be reloaded.
[0990:0268][2016-05-03T13:53:26]i000: Loading prerequisite bootstrapper application because managed host could not be loaded, error: 0x80131700.
```
- Describe the behavior you expected and how it differed from the actual behavior.

> Prerequisite package installation should stop when a machine restart is required and
> 1. provide a message that a machine restart is required and installation must be restarted by the user once the machine resumed when burn is run in full display mode
> 2. a meaningful error code should be returned in all cases
## Feature requests

If this issue is a feature request:
- Describe the scenario and benefits that the feature supports.

> (Provide lots of specific detail so we understand the context around the feature.)
- Describe how you're accomplishing the feature today (if possible).

> (Be as specific as you can and err on the side of providing too much information.)
- Describe what you'd like the new feature to do.

> (There's no need to design it but suggestions are appreciated.)
## Triage

Issues are triaged at [online meetings](https://www.firegiant.com/blog/), generally held alternate Tuesdays at 9:30 a.m. Pacific time (UTC-7/UTC-8). Meeting requests are sent to the [wix-devs](http://wixtoolset.org/documentation/mailinglist/#wix-devs) mailing list.
## Discussions and support

Note that discussions are best held on the mailing lists, not in issues. Please do not open issues requesting support or debugging help; first start a thread on the [wix-users](http://wixtoolset.org/documentation/mailinglist/#wix-users) mailing list and open an issue if the discussion suggests a bug in the WiX Toolset or that a feature request is appropriate.

**Comment by dsed** [comment]
[Logs.zip](https://github.com/wixtoolset/issues/files/246980/Logs.zip)

**Comment by dsed** [comment]
Attached the log files

**Comment by robmen** [comment]
>  Please do not open issues requesting support or debugging help; first start a thread on the wix-users mailing list and open an issue if the discussion suggests a bug in the WiX Toolset or that a feature request is appropriate.

Your BA is likely failing to load causing the problem, please go get support on mailing list.

**Comment by dsed** [comment]
I am neither requesting support nor debugging help. I am reporting a WiX issue. When running the BA without /norestart the prerequisite BA restarts the machine after upgrading .Net and my managed BA runs like a charm. But when specifying /norestart the restart request gets completely ignored in the prerequisite BA, instead it tries to load my managed BA and of course this will fail, because it requires .Net 4.5.2 instead of .Net 4.5.1. When you might have a closer look at the log files you would see that burn is still detecting .Net 4.5.1 on the system while looping and re-installs .Net 4.5.2 again and again.

**Comment by barnson** [comment]
It looks like WixStdBA isn't looking at the restart flag when determining success of a pre-req.

**Comment by robmen** [comment]
@dsed, my bad. You didn't follow the instructions in the template so that's usually an indicator the issue isn't a real issue so I only start skimming the report and a quick look seemed like a support request.

**Comment by rseanhall** [comment]
Supporting /norestart in full display mode will require adding the Success page in mbapreq.  In `OnShutdown`, WixStdBA should only attempt to reload the bootstrapper if the restart result is `NONE`.

**Comment by hugo972** [comment]
I have the same exact issue with Wix 3.11 while installing .NET 4.6.2 in silent mode. Any updates regarding this issue?

We were stuck in the apply .NET package stage because the .NET setup returned 3010 which cause Wix to retry its installation (I assumed because of this code wix3-develop\src\burn\engine\apply.cpp:1690). Only after we converted 3010 return code from .NET installation exit code into forceReboot we got into the state described in this issue where the BA is shutting down and restarting. 

@dsed did you also had to use exit code behavior to make it work (before you got into the BA infinite loop state)?
